src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._form = this._popupElement.querySelector(".modal__form");
+    this._deleteSubmitBtn = this._popupElement.querySelector("#confirm-button");
+  }
+
+  setSubmitAction(handler) {
+    this._handleSubmit = handler;
+  }
+
+  setDeleteLoading(loading) {

```

Comment:
You've implemented the logic for rendering loading via the class methods. Excellent!